### PR TITLE
Add HelpText examples to docs

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 1]
```diff
@@ -171,6 +171,30 @@ By default, the label is positioned above the SearchField. The `labelPosition` p
 <SearchField label="Search" labelPosition="side" labelAlign="end" />
 ```
 
+### Help text
+[View guidelines](https://spectrum.adobe.com/page/help-text/#Usage-guidelines)
```
Comment (ktabors) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx:175]: I noticed their docs show that for error messages and the API has 'showErrorIcon`. I think one of these examples should have that too.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 2]
```diff
@@ -171,6 +171,30 @@ By default, the label is positioned above the SearchField. The `labelPosition` p
...
+
+  return (
+    <SearchField
+      validationState={isValid ? 'valid' : 'invalid'}
+      value={value}
+      onChange={setValue}
+      label="Search your favorite number"
+      maxLength={1}
+      description="Enter a single digit number."
+      errorMessage={value === '' ? 'Empty input not allowed.' : 'Single digit numbers are 0-9.'}
```
Comment (ktabors) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx:192]: The only way I got the second error message was to enter a negative sign. Are there other ways? I'm thinking about usability for people reading the docs.

**Review by ktabors — CHANGES_REQUESTED** [review]
The side labels on these four components touch the input.
Also, all labels are now side labels for these 4.
Should we have the other components that have these too?

**Comment by devongovett** [comment]
@snowystinger NumberField and ComboBox are missing.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx:192]: any letter will do it

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx:175]: correct

**Review by LFDanLu — DISMISSED** [review]
LGTM

File: packages/@react-spectrum/combobox/docs/ComboBox.mdx [hunk 3]
```diff
@@ -839,6 +839,42 @@ For right-to-left (RTL) languages, this is flipped.
 </ComboBox>
 ```
 
+### Help text
+[View guidelines](https://spectrum.adobe.com/page/combo-box/#Help-text-(description-and-error-message))
+
+Both a description and an error message can be supplied to a ComboBox. Description shows all the time except when the `validationState` is "invalid". The error message should help the user fix their input quickly and it should be changed depending on the detected error. Both the description and error message should be localized as well.
```
Comment (dannify) [inline packages/@react-spectrum/combobox/docs/ComboBox.mdx:845]:
```suggestion
Both a description and an error message can be supplied to a ComboBox. The description is always visible unless the `validationState` is “invalid” and an error message is provided. The error message can be used to help the user fix their input quickly and should be specific to the detected error. All strings should be localized.
```

Comment (dannify) [inline packages/@react-spectrum/combobox/docs/ComboBox.mdx:845]: What do you think of this ^?

Comment (snowystinger) [reply packages/@react-spectrum/combobox/docs/ComboBox.mdx:845]: @dannify which do you like more? you gave two suggestions.
I think I like the first one more.

Comment (dannify) [reply packages/@react-spectrum/combobox/docs/ComboBox.mdx:845]: oh weird, not sure what happened there. Should have been the first comment only.

**Review by LFDanLu — APPROVED** [review]

**Review by devongovett — APPROVED** [review]

**Pull request merged**